modules/core/src/main/scala/gem/math/Wavelength.scala

```diff
@@ -4,54 +4,41 @@
 package gem.math
 
 import scalaz.{ Order, Show }
-import scalaz.std.anyVal.intInstance
-import scalaz.syntax.equal._
-
-/**
- * Exact wavelengths represented as unsigned integral angstroms in the range [0 .. Int.MaxValue]
- * which means the largest representable wavelength is 214.7483647 mm.
- * @param toAngstroms This wavelength in integral angstroms (10^-10 of a meter).
- */
-final class Wavelength private (val toAngstroms: Int) {
-
-  // Sanity checks … should be correct via the companion constructor.
-  assert(toAngstroms >= 0, s"Invariant violated. $toAngstroms is negative.")
-  assert(toAngstroms <= Int.MaxValue, s"Invariant violated. $toAngstroms is larger than Int.MaxValue.")
-
-  /** String representation of this Wavelength, for debugging purposes only. */
-  override def toString =
-    f"Wavelength($toAngstroms Å)"
-
-  /** Wavelengths are equal if their magnitudes are equal. */
-  override def equals(a: Any) =
-    a match {
-      case a: Wavelength => a.toAngstroms === toAngstroms
-      case _             => false
-    }
-
-  override def hashCode =
-    toAngstroms
-
-}
+import spire.implicits._
+import spire.math.UInt
+import libra._
 
 object Wavelength {
+  implicit final class BaseQuantityOps[A](val a: A) extends AnyVal {
+    /** Syntax to create a quantity using the Å symbol */
+    def Å: QuantityOf[A, Wavelength, Ångström] = Quantity(a)
+  }
+
+  /** Construct a WavelengthInÅngström from unsigned integer angstroms. */
+  def fromAngstroms(angstroms: UInt): WavelengthInÅngström =
+    angstroms.Å
 
-  final lazy val ZeroAngstroms = unsafeFromAngstroms(0)
+}
+
+object WavelengthInÅngström {
```

Comment:
Yeah, it is the correct way to write bit it is very hard to write and could create problems. I'm open to write it in ASCII